source/Test.cpp

```diff
@@ -192,12 +274,81 @@ void Test::Step(Context &context, UI &menuPanels, UI &gamePanels, PlayerInfo &pl
 		return;
 	}
 	
-	const TestStep &stepToRun = steps[context.stepToRun];
+	// All processing was done just before this step started.
+	context.unprocessedInput = false;
+	context.branchesSinceGameStep.clear();
 	
-	// Exit with error on a failing testStep.
-	const string &stepTypeName = STEPTYPE_TO_TEXT.at(stepToRun.stepType);
-	
-	Fail("Test step " + to_string(context.stepToRun) + " (" + stepTypeName + ") failed");
+	bool continueGameLoop = false;
+	while(context.stepToRun < steps.size() && !continueGameLoop)
+	{
```

Comment:
This should be a do-while construct (since we just bounds-checked stepToRun against steps.size() before entering it).